### ISSUE TITLE
`cargo fmt` fails with `half_open_range_patterns`

### ISSUE THREAD

**Issue by JohnTitor** [rust-lang/rustfmt#4009]
This example compiles fine with latest nightly, but `cargo fmt` fails. The failure also occurs on clippy's CI (https://travis-ci.com/rust-lang/rust-clippy/jobs/274511513).
Probably we need rustup and update submodule on rust-lang/rust?

```rust
#![feature(half_open_range_patterns)]
#![feature(exclusive_range_pattern)]

fn main() {
    match 54 {
        5.. => println!("yay!"),
        _ => (),
    }
}
```

error log:
```
error: `X..` range patterns are not supported
 --> main.rs:6:9
  |
6 |         5.. => println!("yay!"),
  |         ^^^ help: try using the maximum value for the type: `5..MAX`
```

**Comment by calebcartwright** [comment]
Yeah I believe the underlying problem is that ATM rustfmt is using a pretty old version of the rustc internals (v610 of the rustc-ap-* crates, including the one for libsyntax).

The refactoring/splitting of libsyntax over the last couple months has created some challenges in upgrading the rustc-ap-* versions within rustfmt, but I think that all of our blockers will be removed once https://github.com/rust-lang/rust/pull/68135 is merged (and then gets published in rustc-ap)

**Comment by calebcartwright** [comment]
@topecongiro - this can be closed with #4043 being merged
